internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -34,62 +34,223 @@
 package metadata_manager
 
 import (
+	"encoding/json"
+	"path/filepath"
+	"strconv"
+	"time"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
 	"github.com/Azure/azure-storage-fuse/v2/internal/dcache"
 )
 
 // BlobMetadataManager is the implementation of MetadataManager interface
 type BlobMetadataManager struct {
-	cacheDir string
+	cacheDir         string
+	storageCallbacks dcache.StorageCallbacks
 }
 
 // NewMetadataManager creates a new implementation of the MetadataManager interface
-func NewMetadataManager(cacheDir string) (MetadataManager, error) {
+func NewMetadataManager(cacheDir string) (*BlobMetadataManager, error) {
 	return &BlobMetadataManager{
 		cacheDir: cacheDir,
 	}, nil
 }
 
 // CreateFileInit creates the initial metadata for a file
 func (m *BlobMetadataManager) CreateFileInit(filePath string, fileMetadata *dcache.FileMetadata) error {
-	// Dummy implementation
-	return nil
+	// Convert the metadata to JSON
+	jsonData, err := json.MarshalIndent(fileMetadata, "", "  ")
+	if err != nil {
+		log.Debug("CreateFileInit :: Failed to marshal metadata to JSON: %v", err)
+		return err
+	}
+	// Store the open-count in the metadata blob property
+	openCount := "0"
+	metadata := map[string]*string{
+		"opencount": &openCount,
+	}
+
+	err = m.storageCallbacks.PutBlobInStorage(internal.WriteFromBufferOptions{
+		Name:                   filepath.Join(m.cacheDir, "Objects", filePath),
+		Metadata:               metadata,
+		Data:                   jsonData,
+		IsNoneMatchEtagEnabled: true,
+		EtagMatchConditions:    "",
+	})
+	if err != nil {
+		log.Debug("CreateFileInit :: Failed to put blob in storage: %v", err)
```

Comment:
error log, and log the file path too.
also we should distinguish between failures due to if-none-match condition or other errors.
The former can be a warning and latter would be error.